.github/workflows/ci.yml

```diff
@@ -0,0 +1,43 @@
+name: CI
+
+on:
+  push:
+    branches: [master]
+    tags: ["*"]
+  pull_request:
+
+env:
+  BINARYBUILDER_AUTOMATIC_APPLE: true
+
+jobs:
+  test:
+    runs-on: [self-hosted]
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          # Add a job that uses the privileged builder with squashfs shards
+          - env:
+              - BINARYBUILDER_RUNNER: privileged
+              - BINARYBUILDER_USE_SQUASHFS: true
+
+          # Add a job that uses the unprivileged builder with unpacked shards
+          - env:
+              - BINARYBUILDER_RUNNER: unprivileged
+              - BINARYBUILDER_USE_SQUASHFS: false
+
+          # Add a job that uses the docker builder with unpacked shards
+          - env:
+              - BINARYBUILDER_RUNNER: docker
+              - BINARYBUILDER_USE_SQUASHFS: false
+    steps:
+      - uses: actions/checkout@v2
+      - uses: julia-actions/julia-buildpkg@latest
+      # - name: Setup git user
+      #   run: |
+      #     git config --global user.name "Binary Builder"
+      #     git config --global user.email "binary.builder@julialang.org"
+      - uses: julia-actions/julia-runtest@latest
+      - uses: julia-actions/julia-uploadcodecov@latest
```

Comment:
That action is deprecated. It's better to use julia-actions/julia-processcoverage and codecov/codecov-action instead. That also doesn't require the token anymore. See pkgtemplates for an example: https://github.com/invenia/PkgTemplates.jl/blob/5cf93cc4d8ed5b0b6887982046066fec238774c2/test/fixtures/AllPlugins/.github/workflows/ci.yml#L40